src/sql/workbench/contrib/scripting/browser/scriptingActions.ts

```diff
@@ -311,34 +311,18 @@ CommandsRegistry.registerCommand({
 CommandsRegistry.registerCommand({
 	id: OE_REFRESH_COMMAND_ID,
 	handler: async (accessor, args: ObjectExplorerActionsContext): Promise<void> => {
-		const connectionManagementService = accessor.get(IConnectionManagementService);
-		const capabilitiesService = accessor.get(ICapabilitiesService);
 		const objectExplorerService = accessor.get(IObjectExplorerService);
 		const logService = accessor.get(ILogService);
 		const notificationService = accessor.get(INotificationService);
-		const connection = new ConnectionProfile(capabilitiesService, args.connectionProfile);
-		if (connectionManagementService.isConnected(undefined, connection)) {
-			let treeNode = await getTreeNode(args, objectExplorerService);
-			if (!treeNode) {
-				await objectExplorerService.updateObjectExplorerNodes(connection.toIConnectionProfile());
-				treeNode = objectExplorerService.getObjectExplorerNode(connection);
-			}
-			if (treeNode) {
-				const tree = objectExplorerService.getServerTreeView().tree;
-				try {
-					await tree.collapse(treeNode);
-					await objectExplorerService.refreshTreeNode(treeNode.getSession(), treeNode);
-					await tree.refresh(treeNode);
-					await tree.expand(treeNode);
-				} catch (err) {
-					// Display message to the user but also log the entire error to the console for the stack trace
-					notificationService.error(localize('refreshError', "An error occurred refreshing node '{0}': {1}", args.nodeInfo.label, getErrorMessage(err)));
-					logService.error(err);
-				}
-
-			} else {
-				logService.error(`Could not find tree node for node ${args.nodeInfo.label}`);
-			}
+		const treeNode = await getTreeNode(args, objectExplorerService);
```

Comment:
why don't we need to null check this as we were before?

Reply:
These are just for specific nodes (scriptable ones) which should always exist. That's what the other actions in this file assume and it seems reasonable to just not bother having this logic. The fallback logic here was just to get the connection node anyways which I would say isn't what we want to do if we somehow got called for a node that we can't find.